test/unit/auth.js

```diff
@@ -113,4 +115,224 @@ describe('DropboxAuth', () => {
+      dbxAuth.generatePKCECodes();
+      chai.assert.isTrue(!!dbxAuth.codeChallenge);
+    });
+  });
+
+  describe('getAccessTokenFromCode', () => {
+    it('throws an error without a clientID', () => {
+      const dbxAuth = new DropboxAuth();
+      chai.assert.throws(
+        DropboxAuth.prototype.getAccessTokenFromCode.bind(dbxAuth, 'foo', 'bar'),
+        Error,
+        'A client id is required. You can set the client id using .setClientId().',
+      );
+    });
+
+    it('throws an error when not provided a client secret or code challenge', () => {
+      const dbxAuth = new DropboxAuth({ clientId: 'foo' });
+      chai.assert.throws(
+        DropboxAuth.prototype.getAccessTokenFromCode.bind(dbxAuth, 'foo', 'bar'),
+        Error,
+        'You must use PKCE when generating the authorization URL to not include a client secret',
+      );
+    });
+
+    it('sets the right path for fetch request', () => {
+      const dbxAuth = new DropboxAuth({
+        clientId: 'foo',
+        clientSecret: 'bar',
+      });
+
+      const fetchSpy = sinon.spy(dbxAuth, 'fetch');
+      dbxAuth.getAccessTokenFromCode('foo', 'bar');
+      const path = dbxAuth.fetch.getCall(0).args[0];
+      chai.assert.isTrue(fetchSpy.calledOnce);
+      chai.assert.equal('https://api.dropboxapi.com/oauth2/token?grant_type=authorization_code&code=bar&client_id=foo&client_secret=bar&redirect_uri=foo', path);
+    });
+
+    it('sets the right path without a redirect uri', () => {
+      const dbxAuth = new DropboxAuth({
+        clientId: 'foo',
+        clientSecret: 'bar',
+      });
+
+      const fetchSpy = sinon.spy(dbxAuth, 'fetch');
+      dbxAuth.getAccessTokenFromCode(undefined, 'bar');
+      const path = dbxAuth.fetch.getCall(0).args[0];
+      chai.assert.isTrue(fetchSpy.calledOnce);
+      chai.assert.equal('https://api.dropboxapi.com/oauth2/token?grant_type=authorization_code&code=bar&client_id=foo&client_secret=bar', path);
+    });
+
+    it('sets the correct headers for fetch request', () => {
+      const dbxAuth = new DropboxAuth({
+        clientId: 'foo',
+        clientSecret: 'bar',
+      });
+
+      const fetchSpy = sinon.spy(dbxAuth, 'fetch');
+      dbxAuth.getAccessTokenFromCode('foo', 'bar');
+      const { headers } = dbxAuth.fetch.getCall(0).args[1];
+      chai.assert.isTrue(fetchSpy.calledOnce);
+      chai.assert.equal(headers['Content-Type'], 'application/x-www-form-urlencoded');
+    });
+  });
+
+  describe('checkAndRefreshAccessToken', () => {
+    it('does not refresh without refresh token or clientId', () => {
+      const dbxAuth = new DropboxAuth();
+      const refreshSpy = sinon.spy(dbxAuth, 'refreshAccessToken');
+      dbxAuth.checkAndRefreshAccessToken();
+      chai.assert.isTrue(refreshSpy.notCalled);
+    });
+
+    it('doesnt refresh token when not past expiration time', () => {
+      const dbxAuth = new DropboxAuth({
+        accessTokenExpiresAt: new Date(2100, 11, 19),
```

Comment:
nit: better practice is to take current date and add time so that if we reach this date, we don't just see tests randomly failing for no obvious reason